tests/subsys/bluetooth/fast_pair/storage/factory_reset/src/main.c

```
@@ -0,0 +1,470 @@
+			  "Invalid order of elements in section");
+
+	mark_test_as_finished(RESET_TEST_FINISHED_SECTION_PLACEMENT_BIT_POS);
+}
+
+static int settings_validate(settings_load_direct_cb cb)
+{
+	int err;
+	int err_cb = 0;
+
+	err = settings_load_subtree_direct(NULL, cb, &err_cb);
+	if (err) {
+		return err;
+	}
+
+	if (err_cb) {
+		return err_cb;
+	}
+
+	return 0;
+}
+
+static int settings_validate_empty_cb(const char *key, size_t len, settings_read_cb read_cb,
+				      void *cb_arg, void *param)
+{
+	int *err = param;
+
+	*err = -EFAULT;
+	return *err;
+}
+
+static int fp_settings_data_validate_empty_cb(const char *key, size_t len, settings_read_cb read_cb,
+					      void *cb_arg, void *param)
+{
+	int *err = param;
+
+	if (!strncmp(key, SETTINGS_RESET_TEST_STAGE_FULL_NAME,
+		     sizeof(SETTINGS_RESET_TEST_STAGE_FULL_NAME))) {
+		return 0;
+	}
+
+	if (!strncmp(key, SETTINGS_RESET_IN_PROGRESS_FULL_NAME,
+		     sizeof(SETTINGS_RESET_IN_PROGRESS_FULL_NAME))) {
+		int rc;
+		bool reset_in_progress;
+
+		if (len != sizeof(reset_in_progress)) {
+			*err = -EINVAL;
+			return *err;
+		}
+
+		rc = read_cb(cb_arg, &reset_in_progress, len);
+		if (rc < 0) {
+			*err = rc;
+			return *err;
+		}
+
+		if (rc != len) {
+			*err = -EINVAL;
+			return *err;
+		}
+
+		if (reset_in_progress) {
+			*err = -EFAULT;
+			return *err;
+		}
+
+		return 0;
+	}
+
+	*err = -EFAULT;
+	return *err;
+}
+
+static void fp_ram_clear(void)
+{
+	fp_storage_ak_ram_clear();
+	fp_storage_pn_ram_clear();
+	fp_storage_manager_ram_clear();
+}
+
+static void setup_fn(void)
+{
+	int err;
+
+	err = settings_validate(settings_validate_empty_cb);
+	zassert_ok(err, "Settings have not been fully deleted");
+
+	err = settings_load();
+	zassert_ok(err, "Unexpected error in settings load");
+}
+
+static int settings_get_name_cb(const char *key, size_t len, settings_read_cb read_cb, void *cb_arg,
+				void *param)
+{
+	char *name = param;
+
+	strncpy(name, key, SETTINGS_MAX_NAME_LEN);
+	return -EALREADY;
```

Should we actually return an error here? Currently it works, because the error code returned here is not propagated further.